boa_engine/src/object/mod.rs

```diff
@@ -1773,58 +1772,119 @@ impl Object {
         self.properties.remove(key)
     }
 
+    /// Check if a private name exists.
```

Comment:
I'm seeing this resembles the [PrivateElementFind](https://tc39.es/ecma262/#sec-privateelementfind) operation and friends. Could this be moved to `boa_engine/src/object/operations.rs` and adapted to somewhat resemble the spec operations? Probably also `initialize_instance_elements`.

I say this because there's a host hook [HostEnsureCanAddPrivateElement](https://tc39.es/ecma262/#sec-hostensurecanaddprivateelement) that needs to be called on the [PrivateFieldAdd](https://tc39.es/ecma262/#sec-privatefieldadd) and [PrivateMethodOrAccessorAdd](https://tc39.es/ecma262/#sec-privatemethodoraccessoradd) operations to restrict the addition of new private fields for an object.

Reply:
Good call. I've refactored those methods and they are mostly equivalent to the spec now.